modules/programs/pet.nix

```
@@ -0,0 +1,146 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.pet;
+
+  snippetType = types.submodule {
+    options = {
+      description = mkOption {
+        type = types.str;
+        default = "";
+        example = "Show expiration date of SSL certificate";
+        description = ''
+          description of the snippet
+        '';
+      };
+
+      command = mkOption {
+        type = types.str;
+        default = "";
+        example =
+          "echo | openssl s_client -connect example.com:443 2>/dev/null |openssl x509 -dates -noout";
+        description = ''
+          the command
+        '';
+      };
+
+      output = mkOption {
+        type = types.str;
+        default = "";
+        example = ''
+          notBefore=Nov  3 00:00:00 2015 GMT
+          notAfter=Nov 28 12:00:00 2018 GMT
+        '';
+        description = ''
+          example output of the command
+        '';
+      };
+    };
+  };
+
+  mkSnippetConfigPart = snippet:
+    let
+      esc = escape [ ''"'' ];
+      desc = esc snippet.description;
+      comm = esc snippet.command;
+      out = esc snippet.output;
+    in ''
+      [[snippets]]
+        description = "${desc}"
+        command = "${comm}"
+        output = "${out}"
+    '';
+
+in {
+  options.programs.pet = {
+    enable = mkEnableOption "pet";
+
+    snippetfile = mkOption {
+      type = types.str;
+      default = "${config.xdg.configHome}/pet/snippet.toml";
```

Need to add a `defaultText` field, otherwise the manual will have a build dependency on the current configuration.
```suggestion
      default = "${config.xdg.configHome}/pet/snippet.toml";
      defaultText = "\${config.xdg.configHome}/pet/snippet.toml";
```